### ISSUE TITLE
Error in demangling "_ZZN17TestLargestRegion18TestNonRectangularEvENUt_D2Ev"

### ISSUE THREAD

**Issue by calixteman** [gimli-rs/cpp_demangle#157]
When I try to demangle this string, I get:
thread 'main' panicked at 'a Display implementation return an error unexpectedly: Error', libcore/result.rs:945:5

c++filt is ok and returns:
TestLargestRegion::TestNonRectangular()::{unnamed type#1}::~TestNonRectangular()

**Comment by fitzgen** [comment]
Thanks for filing an issue!

**Comment by miguelsaldivar** [comment]
I'll try and fix this :smile:

**Comment by fitzgen** [comment]
Hey @miguelsaldivar -- saw your question on IRC.

You can enable the debug logging by building with the "logging" feature: https://github.com/gimli-rs/cpp_demangle/blob/master/Cargo.toml#L56-L58

For example:

```
cargo run --example cppfilt --features logging -- _Z3fooiPc
```

**Comment by miguelsaldivar** [comment]
@fitzgen thanks for the response, I'll work on this some more this weekend :+1:

**Comment by fitzgen** [comment]
In general, we want to match `libiberty` (which is part of binutils, and is what `c++filt` uses). But particularly in cases where different demanglers don't agree, we want to make sure that we are comparing with the latest `libiberty`, so I'd suggest building it and c++filt from source.

https://www.gnu.org/software/binutils/

**Comment by miguelsaldivar** [comment]
Okay, I'll build it locally and give it a try. If I run into any issues, where is the best place to contact you? I don't want to spam this issue.

**Comment by fitzgen** [comment]
This issue is a fine place to ask questions :)

**Comment by miguelsaldivar** [comment]
```
AST = Encoding(
    Function(
        Local(
            Relative(
                Function(
                    Nested(
                        Unqualified(
                            CvQualifiers {
                                restrict: false,
                                volatile: false,
                                const_: false
                            },
                            None,
                            BackReference(
                                0
                            ),
                            Source(
                                SourceName(
                                    Identifier {
                                        start: 25,
                                        end: 43
                                    }
                                )
                            )
                        )
                    ),
                    BareFunctionType(
                        [
                            Builtin(
                                Standard(
                                    Void
                                )
                            )
                        ]
                    )
                ),
                Some(
                    Nested(
                        Unqualified(
                            CvQualifiers {
                                restrict: false,
                                volatile: false,
                                const_: false
                            },
                            None,
                            BackReference(
                                1
                            ),
                            CtorDtor(
                                BaseConstructor
                            )
                        )
                    )
                ),
                None
            )
        ),
        BareFunctionType(
            [
                Builtin(
                    Standard(
                        Void
                    )
                )
            ]
        )
    )
)
```
With the AST looking like this, how can I get the `SourceName` from `CtorDtor::Demangle`?

**Comment by miguelsaldivar** [comment]
And I ask this because when getting to `CtorDtor::Demangle` the leaf type is `UnnamedType`, and that doesn't have a name.

**Comment by miguelsaldivar** [comment]
Closed with #168
